src/notifier/service/transaction.go

```
@@ -4,35 +4,31 @@ import (
 	"bytes"
 	"context"
 
-	"github.com/consensys/orchestrate/pkg/errors"
-	"github.com/consensys/orchestrate/pkg/toolkit/app/log"
 	"github.com/consensys/orchestrate/src/entities"
 	"github.com/consensys/orchestrate/src/infra/api"
 	usecases "github.com/consensys/orchestrate/src/notifier/notifier/use-cases"
 	"github.com/consensys/orchestrate/src/notifier/service/types"
 )
 
-var TransactionMessageType entities.RequestMessageType = "transaction_notification"
+const TransactionMessageType entities.RequestMessageType = "transaction_notification"
 
 type TransactionHandler struct {
 	sendUC     usecases.SendNotificationUseCase
 	maxRetries int
-	logger     *log.Logger
 }
 
 func NewTransactionHandler(sendUC usecases.SendNotificationUseCase, maxRetries int) *TransactionHandler {
 	return &TransactionHandler{
 		sendUC:     sendUC,
 		maxRetries: maxRetries,
-		logger:     log.NewLogger().SetComponent(messageListenerComponent),
 	}
 }
 
 func (mch *TransactionHandler) HandleTransactionReq(ctx context.Context, msg *entities.Message) error {
 	req := &types.TransactionMessageRequest{}
 	err := api.UnmarshalBody(bytes.NewReader(msg.Body), req)
 	if err != nil {
-		return errors.InvalidFormatError("invalid transaction request type")
+		return err
```

we should always parse raw errors into internal type errors

---

as you can see, the err is lost here. We should return the actual error. This is correct in HTTP and in Kafka as well because it's a 4xx error.